launcher/ui/pages/global/APIPage.cpp

```diff
@@ -165,8 +174,34 @@ void APIPage::applySettings()
     if (!metaURL.isEmpty() && metaURL.scheme() == "http") {
         metaURL.setScheme("https");
     }
-
     s->set("MetaURLOverride", metaURL.toString());
+
+    QUrl resourceURL(ui->resourceURL->text());
+    // Add required trailing slash
+    if (!resourceURL.isEmpty() && !resourceURL.path().endsWith('/')) {
+        QString path = resourceURL.path();
+        path.append('/');
+        resourceURL.setPath(path);
+    }
+    // HTTP may not be allowed either?
+    if (!resourceURL.isEmpty() && resourceURL.scheme() == "http") {
+        resourceURL.setScheme("https");
+    }
```

Comment:
I see this code reused multiple times can we make it a function that gets a QUrl and returns a QString(it can even be a lambda function)